### ISSUE TITLE
Path parameters with dots generates nonfunctional typescript

### ISSUE THREAD

**Issue by ianrose14** [acacode/swagger-typescript-api#413]
## Summary
If a path parameter has a dot in it (e.g. `item.id`) then `swagger-typescript-api` will rewrite this to camelCase for the _parameter name_ but not in the path string interpolation, which leads in nonfunctional code.

## Repro
Here's a minimal (or at least small) input to reproduce this:
```json
{
  "swagger": "2.0",
  "info": {
    "title": "unset",
    "version": "unset"
  },
  "consumes": ["application/json"],
  "produces": ["application/json"],
  "paths": {
    "/foobar/{truck.id}/item": {
      "post": {
        "summary": "Repros an issue",
        "operationId": "ReproFunc",
        "responses": {
          "default": {
            "description": "A response.",
            "schema": {
              "$ref": "#/definitions/myResponse"
            }
          }
        },
        "parameters": [
          {
            "name": "truck.id",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ]
      }
    }
  },
  "definitions": {
    "myResponse": {
      "type": "object"
    }
  }
}
```

Expected output:
```typescript
  foobar = {
    /**
     * No description
     *
     * @name ReproFunc
     * @summary Repros an issue
     * @request POST:/foobar/{truck.id}/item
     */
    reproFunc: (truckId: string, params: RequestParams = {}) =>
      this.request<any, MyResponse>({
        path: `/foobar/{truckId}/item`,
        method: "POST",
        ...params,
      }),
  };
```

Actual output:
```typescript
  foobar = {
    /**
     * No description
     *
     * @name ReproFunc
     * @summary Repros an issue
     * @request POST:/foobar/{truck.id}/item
     */
    reproFunc: (truckId: string, params: RequestParams = {}) =>
      this.request<any, MyResponse>({
        path: `/foobar/{truck.id}/item`,
        method: "POST",
        ...params,
      }),
  };
```

Note that the path (`/foobar/{truck.id}/item`) references a variable ("truck.id") that does not exist, since the function parameter was rewritten to be `truckId`.

p.s. The obvious solution here is "don't use parameters with dots in them - that's weird!".  Unfortunately the swagger we are using is _itself_ generated by [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway) which supports nested fields via dot notation ([example](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/a_bit_of_everything.proto#L696)) so there's no easy way to avoid this...

**Comment by js2me** [comment]
Hi @ianrose14 ! Thanks for this issue, will fix it soon

**Comment by js2me** [comment]
@ianrose14 should be fixed in 11.0.0 release.   
Please reopen this issue if problem is still occurring
